lua/resession/init.lua

```diff
@@ -69,11 +69,12 @@ M.get_current = function()
   return tab_sessions[tabpage] or current_session
 end
 
----Detach from the current session
+---Detach from the current session, emitting the `ResessionDetachPost` User autocmd upon completion
 M.detach = function()
   current_session = nil
   local tabpage = vim.api.nvim_get_current_tabpage()
   tab_sessions[tabpage] = nil
+  require("resession.util").event("DetachPost")
```

Comment:
I think this should only be dispatched if there was previously a session attached. Also, it would be nice to include some data in the event payload. `session_name` and `is_tab_session` come to mind. Those probably make sense to add to all of the event dispatches

Reply:
I'm not sure how this sort of information can be included with an autocommand event. Or do you mean passing that data into the hook callback as an argument?